### PR TITLE
[rpc] rename init_model_parallel to init_rpc

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29762 [rpc] rename init_model_parallel to init_rpc**

Rename this API as discussed, since its use cases extend beyond only
model parallelism.

Differential Revision: [D18491743](https://our.internmc.facebook.com/intern/diff/D18491743/)

### CODE REVIEW

File: torch/distributed/rpc/backend_registry.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def process_group_init_backend_handler(
     # Initialize ProcessGroup.
     if dist.is_initialized():
         raise RuntimeError(
-            "Default process group must not be initialized before `init_model_parallel`."
```
Comment (rohan-varma) [inline torch/distributed/rpc/backend_registry.py]: removed backticks

**Review by pritamdamania87 — APPROVED** [review]

File: torch/distributed/rpc/__init__.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ def is_available():
     from .api import *  # noqa: F401
     import torch.distributed.autograd
 
-    def init_model_parallel(
+    def init_rpc(
         self_name,
         backend=backend_registry.BackendType.PROCESS_GROUP,
```
Comment (pritamdamania87) [inline torch/distributed/rpc/__init__.py]: We mention `model parallel` in the doc, should we rename that too?

File: test/rpc_test.py [hunk 3]
```diff
@@ -478,7 +478,7 @@ def test_sync_rpc(self):
     @dist_init(setup_model_parallel=False)
```
Comment (pritamdamania87) [inline test/rpc_test.py]: Should this flag be renamed too?

Comment (rohan-varma) [reply test/rpc_test.py]: Good catch.

File: torch/distributed/rpc/__init__.py (hunk 2 above)
Comment (rohan-varma) [reply torch/distributed/rpc/__init__.py]: Yep, instead of " Initializes model parallel primitives" I will change it to " Initializes rpc  primitives"

**Review by zhaojuanmao — COMMENTED** [review]
should we rename _init_rpc() as well?

**Comment by rohan-varma** [comment]
> should we rename _init_rpc() as well?

I think we can rename it `_init_rpc_backend` since it sets up things like the process group backend.

**Review by mrshenli — APPROVED** [review]

File: torch/distributed/rpc/__init__.py [hunk 4]
```diff
@@ -31,7 +31,7 @@ def init_model_parallel(
         rpc_timeout=DEFAULT_RPC_TIMEOUT,
     ):
         r"""
-        Initializes model parallel primitives such as the local rpc agent
+        Initializes rpc primitives such as the local rpc agent
```
Comment (pietern) [inline torch/distributed/rpc/__init__.py]: Nit for later fixup: capitalize RPC.

**Pull request closed without merging**